ui/app/routes/access-control/roles/new.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Route from '@ember/routing/route';
+import { inject as service } from '@ember/service';
+
+export default class AccessControlRolesNewRoute extends Route {
+  @service can;
+  @service router;
+
+  beforeModel() {
+    if (this.can.cannot('write role')) {
+      this.router.transitionTo('/access-control/roles');
+    }
+  }
+
+  async model() {
+    let role = await this.store.createRecord('role', {
+      name: '',
+    });
+    return {
+      role,
+      policies: await this.store.findAll('policy'),
+    };
+  }
+
+  resetController(controller, isExiting) {
+    if (isExiting) {
+      // If user didn't save, delete the freshly created model
+      if (controller.model.role.isNew) {
+        controller.model.role.destroyRecord();
+      }
+    }
+  }
```

Comment:
When we hit /new in our new routes, we first generate a new, empty, instance of the resource we're attempting to create (in the browser only; nothing saved until explicitly done by the user).

We can then populate it by way of the form in the UI

If the user leaves before saving the new thing, `resetController` destroys the record so that there aren't garbage collection issues later on.